File: matiasmasso/Mat.NET/DT2/DTONoticia.vb
Public Class DTONoticia
    Inherits DTONoticiaBase

    Property product As DTOProduct
    Property priority As PriorityLevels

    Public Sub New()
        MyBase.New(Srcs.News)
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Public Enum PriorityLevels
        Low
        High
    End Enum

    Public Enum wellknowns
        NotSet
        iMat2
    End Enum

    Shared Function wellknown(id As DTONoticia.wellknowns) As DTONoticia
        Dim retval As DTONoticia = Nothing
        Select Case id
            Case DTONoticia.wellknowns.iMat2
                retval = New DTONoticia(New Guid("12C5299A-573C-4A90-860F-3CDA98A713AD"))
        End Select
        Return retval
    End Function


    Shared Function ExcerptOrCroppedText(oNoticia As DTONoticia) As DTOLangText
        Dim retval As New DTOLangText
        If oNoticia.Excerpt IsNot Nothing Then
            If Not oNoticia.Excerpt.IsEmpty Then
                retval = oNoticia.Excerpt
            End If
        End If

        If retval.IsEmpty AndAlso oNoticia.Text IsNot Nothing Then

            Dim sText As String = oNoticia.Text.Esp
            If sText > "" Then

                Dim iMore As Integer = sText.IndexOf("<more/>")
                If iMore < 0 Then iMore = 255
                retval.Esp = sText.Substring(0, iMore)

                sText = oNoticia.Text.Cat
                If sText.Length > 0 Then
                    iMore = sText.IndexOf("<more/>")
                    If iMore < 0 Then iMore = 255
                    retval.Cat = sText.Substring(0, iMore)
                End If

                sText = oNoticia.Text.Eng
                If sText.Length > 0 Then
                    iMore = sText.IndexOf("<more/>")
                    If iMore < 0 Then iMore = 255
                    retval.Eng = sText.Substring(0, iMore)
                End If

                sText = oNoticia.Text.Por
                If sText.Length > 0 Then
                    iMore = sText.IndexOf("<more/>")
                    If iMore < 0 Then iMore = 255
                    retval.Por = sText.Substring(0, iMore)
                End If
            End If
        End If

        Return retval
    End Function

    Shared Function Filtered(oNoticia As DTONoticia, oUser As DTOUser) As DTOLangText
        Dim retval As DTOLangText = Nothing
        If oUser Is Nothing Then
            retval = DTOLangText.Replace(oNoticia.Text, "@UserGuid", "")
        Else
            retval = DTOLangText.Replace(oNoticia.Text, "@UserGuid", oUser.Guid.ToString())
        End If
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTONoticia.cs
using System;

namespace DTO
{
    public class DTONoticia : DTONoticiaBase
    {
        public DTOProduct product { get; set; }
        public PriorityLevels priority { get; set; }
        public DTOPostComment.TreeModel Comments { get; set; }

        public const int IMAGEWIDTH = 325;
        public const int IMAGEHEIGHT = 205;

        public DTONoticia() : base(Srcs.News)
        {
        }

        public DTONoticia(Guid oGuid) : base(oGuid)
        {
        }

        public enum PriorityLevels
        {
            Low,
            High
        }

        public new enum Wellknowns
        {
            NotSet,
            iMat2
        }

        public static DTONoticia Factory(DTOUser oUser)
        {
            DTONoticia retval = new DTONoticia();
            retval.UsrLog = DTOUsrLog.Factory(oUser);
            return retval;
        }

        public static DTONoticia Wellknown(DTONoticia.Wellknowns id)
        {
            DTONoticia retval = null;
            switch (id)
            {
                case DTONoticia.Wellknowns.iMat2:
                    {
                        retval = new DTONoticia(new Guid("12C5299A-573C-4A90-860F-3CDA98A713AD"));
                        break;
                    }
            }
            return retval;
        }

        public new DTOUrl Url()
        {
            DTOUrl retval = DTOUrl.Factory("news");
            retval.Path = new DTOLangText(this.urlFriendlySegment);
            retval.FileExtension = ".html";
            return retval;
        }

        public static DTOLangText ExcerptOrCroppedText(DTONoticia oNoticia)
        {
            DTOLangText retval = new DTOLangText();
            if (oNoticia.Excerpt != null)
            {
                if (!oNoticia.Excerpt.isEmpty())
                    retval = oNoticia.Excerpt;
            }

            if (retval.isEmpty() && !oNoticia.Text.isEmpty())
            {
                string sText = oNoticia.Text.Esp;
                if (sText.isNotEmpty())
                {
                    int iMore = sText.IndexOf("<more/>");
                    if (iMore < 0)
                        iMore = 255;
                    retval.Esp = sText.Substring(0, iMore);

                    sText = oNoticia.Text.Cat;
                    if (sText.Length > 0)
                    {
                        iMore = sText.IndexOf("<more/>");
                        if (iMore < 0)
                            iMore = 255;
                        retval.Cat = sText.Substring(0, iMore);
                    }

                    sText = oNoticia.Text.Eng;
                    if (sText.Length > 0)
                    {
                        iMore = sText.IndexOf("<more/>");
                        if (iMore < 0)
                            iMore = 255;
                        retval.Eng = sText.Substring(0, iMore);
                    }

                    sText = oNoticia.Text.Por;
                    if (sText.Length > 0)
                    {
                        iMore = sText.IndexOf("<more/>");
                        if (iMore < 0)
                            iMore = 255;
                        retval.Por = sText.Substring(0, iMore);
                    }
                }
            }

            return retval;
        }

        public static DTOLangText Filtered(DTONoticia oNoticia, DTOUser oUser)
        {
            DTOLangText retval = null/* TODO Change to default(_) if this is not a reference type */;
            if (oUser == null)
                retval = DTOLangText.Replace(oNoticia.Text, "@UserGuid", "");
            else
                retval = DTOLangText.Replace(oNoticia.Text, "@UserGuid", oUser.Guid.ToString());
            return retval;
        }

        public static string urlAllNoticias(DTOWebDomain domain = null)
        {
            if (domain == null) domain = DTOWebDomain.Default();
            return domain.Url("noticias");
        }

        public bool isVisibleOnNewsSitemap(DTOWebDomain domain)
        {
            bool retval = false;
            int lastValidDaysCount = 2;
            bool dateIsInRange = this.UsrLog.FchLastEdited >= DTO.GlobalVariables.Today().AddDays(-lastValidDaysCount);
            retval = (dateIsInRange && this.Visible && !this.professional && this.Title.HasContent(domain));
            return retval;
        }

        //public SyndicationItem Rss(DTOWebDomain domain)
        //{
        //    var lang = domain.DefaultLang();
        //    var retval = new SyndicationItem()
        //    {
        //        Id = this.Guid.ToString(),
        //        Title = SyndicationContent.CreatePlaintextContent(this.Title.Tradueix(lang)),
        //        Content = SyndicationContent.CreateHtmlContent(this.Excerpt.Tradueix(lang)),
        //        PublishDate = this.Fch,
        //        LastUpdatedTime = this.UsrLog.FchLastEdited
        //    };

        //    retval.Links.Add(SyndicationLink.CreateAlternateLink(new Uri(this.Url().CanonicalUrl(lang))));
        //    return retval;
        //}


    }
}
